docs/customization/api.rst

```diff
@@ -95,4 +105,362 @@ The next step is to modify the security configuration in ``config/packages/secur
 
-After these two steps you can start to use endpoints with new prefixes.
+After these two steps you can start to use endpoints with new prefixes
+
+How to customize serialization?
+-------------------------------
+
+Let's say that you want to change the serialized fields in your responses.
+For an example we will use ``Product`` resource and customize its fields.
+
+Adding a field to response
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Let's say that you want to serialize existing field named ``averageRating`` to ``Product`` in admin response so the administrator would be able to check what is the average rating of product.
+
+First let's copy serialization configuration file named ``Product.xml`` from ``%kernel.project_dir%/vendor/sylius/sylius/src/Sylius/Bundle/ApiBundle/Resources/config/serialization/``
+to ``config/serialization/Product.xml``
+
+Then let's find the attribute ``averageRating``:
+
+.. code-block:: xml
+
+    <!--...-->
+    <attribute name="averageRating">
+            <group>shop:product:read</group>
+    </attribute>
+    <!--...-->
+
+
+and add serialization group that is used by endpoint we want to modify
+
+.. tip::
+
+    You can create your own serialization group for every endpoint or use the one out of the box. If you don't know the name of group for endpoint you want to modify,
+    you can find it by searching for your class configuration file in `%kernel.project_dir%/vendor/sylius/sylius/src/Sylius/Bundle/ApiBundle/Resources/config/api_resources``
+    and look for path that you want to modify.
+
+In this case the new ``group`` is called ``admin:product:read``:
+
+.. code-block:: xml
+
+    <!--...-->
+    <attribute name="averageRating">
+            <group>admin:product:read</group>
+            <group>shop:product:read</group>
+    </attribute>
+    <!--...-->
+
+After this change your response should be extended with new field:
+
+.. code-block:: javascript
+
+    {
+        //...
+        "id": 123,
+        "code": "product_code",
+        "variants": [
+            "/api/v2/shop/product-variants/product-variant-0",
+        ],
+        "averageRating": 3,
+        //...
+    }
+
+.. tip::
+
+    Read more about API Platform `serialization groups <https://api-platform.com/docs/core/serialization/#using-serialization-groups>`_
+
+
+We were able to add a field that exists in ``Product`` class, but what if you want to extend it with custom fields?
+Let's customize response now with your custom fields serialized in response.
+
+Adding a custom field to response
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Let's say that you want to add a new field named ``additionalText`` to ``Product``.
+First we need to create a new serializer that will support our ``Product`` resource, but in this case we have a ``ProductNormalizer`` provided from Sylius.
+Unfortunately we cannot use more than one normalizer per resource, hence we will override existing one.
```

Comment:
Do we really need to override the whole service? Won't decoration do the trick?

Reply:
There is some magic from Api-platforms trait `NormalizerAwareTrait` that won't let me decorate it nicely.